### PR TITLE
feat: add cross-profile symbol execution lock

### LINKED ISSUE #127: [API-12] Add cross-profile symbol arbitration and global position lock
## Contexte

Avec la runtime matrix `exchange × profile`, plusieurs schedules Temporal peuvent tourner en parallèle :

```text
bitmart / perpetual / scalper
bitmart / perpetual / scalper_micro
okx / perpetual / scalper
okx / perpetual / scalper_micro
```

Le risque est qu'un même symbole soit validé par deux profils dans la même minute.

Exemple :

```text
cron-mtf-bitmart-scalper-1m
cron-mtf-bitmart-scalper-micro-1m
```

Les deux schedules peuvent traiter BTCUSDT au même moment.

L'idempotence actuelle inclut `strategy_profile` dans la clé de décision :

```text
exchange
market_type
symbol
timeframe
candle_open_ts
side
strategy_profile
strategy_version
```

Donc ces deux décisions sont différentes :

```text
bitmart:perpetual:BTCUSDT:1m:...:long:scalper:v1
bitmart:perpetual:BTCUSDT:1m:...:long:scalper_micro:v1
```

Cela protège contre les doublons du même profil, mais pas contre les doublons entre profils.

Le runner filtre déjà les symboles avec position ouverte ou ordre ouvert, mais ce filtrage se fait au début du run. Si deux schedules démarrent en parallèle avant qu'un ordre soit visible côté exchange/local DB, les deux peuvent passer.

## Objectif

Ajouter un mécanisme d'arbitrage global pour empêcher plusieurs profils d'ouvrir simultanément une position sur le même :

```text
exchange + market_type + symbol
```

Règle métier :

> Pour un même exchange + market_type + symbol, une seule stratégie peut avoir une ouverture en cours ou une position active.

### 1. Clé globale symbole

Créer une clé indépendante du profil :

```text
global_symbol_key = exchange:market_type:symbol
```

Exemples :

```text
bitmart:perpetual:BTCUSDT
okx:perpetual:ETHUSDT
hyperliquid:perpetual:SOLUSDT
```

Cette clé doit être utilisée avant la soumission d'un ordre.

### 2. Lock global atomique

Créer un mécanisme de réservation global avant `OrderIntentManager::reserve()` ou à l'intérieur de `OrderIntentManager`.

Objectif :

> Si un `order_intent` actif existe déjà pour exchange+market_type+symbol, alors bloquer toute nouvelle intention d'un autre profil.

États considérés actifs, à ajuster selon les enums existantes :

```text
DRAFT
VALIDATED
READY_TO_SEND
SENT
SUBMITTED
PARTIALLY_FILLED
FILLED tant que la position n'est pas clôturée
```

États non bloquants :

```text
FAILED
CANCELLED
REJECTED
EXPIRED
CLOSED
EMERGENCY_CLOSED
```

### 3. Politique d'arbitrage configurable

Ajouter une configuration :

```yaml
cross_profile_arbitration:
  enabled: true
  scope: exchange_market_symbol
  priority:
    - scalper
    - scalper_micro
    - regular
  block_lower_priority: true
  block_same_priority: true
  lock_ttl_seconds: 900
```

Pour la première version :

- **first active intent wins**
- Pas de préemption automatique entre profils.

### 4. Raison de skip explicite

Quand une stratégie est bloquée, retourner/logguer :

```json
{
  "status": "skipped",
  "reason": "cross_profile_symbol_locked",
  "lock": {
    "exchange": "bitmart",
    "market_type": "perpetual",
    "symbol": "BTCUSDT",
    "current_profile": "scalper_micro",
    "blocking_profile": "scalper",
    "blocking_order_intent_id": "...",
    "blocking_decision_key": "..."
  }
}
```

Cette raison doit apparaître dans :

- logs `order_journey`,
- `trade_lifecycle_event` si applicable,
- payload de retour TradeEntry si disponible,
- audit/debug.

### 5. Verrouillage DB robuste

Le verrou doit être atomique côté DB et robuste contre deux workers/processes en parallèle.

Recommandation : créer une table dédiée :

```text
symbol_execution_lock
```

Colonnes proposées :

```text
id
exchange
market_type
symbol
status
owner_profile
owner_decision_key
owner_order_intent_id
locked_at
expires_at
released_at
release_reason
payload
created_at
updated_at
```

Index/contrainte :

```text
unique active lock on (exchange, market_type, symbol) where released_at is null
```

### 6. Expiration de sécurité

Ajouter un TTL pour éviter qu'un lock reste bloqué si un worker crash.

Le lock peut être considéré expiré uniquement si :

- pas d'ordre ouvert
- pas de position ouverte
- order_intent non actif

> Ne jamais expirer/libérer un lock si une position est encore ouverte.

### 7. Libération du lock

Libérer le lock quand :

- ordre rejeté
- ordre annulé
- ordre expiré
- position fermée
- emergency close confirmé
- order_intent failed avant soumission

Ne pas libérer si :

- ordre envoyé mais statut inconnu
- position potentiellement ouverte
- SL/protection en cours de vérification
- reconciliation non concluante

### 8. Intégration reconciliation

La réconciliation REST/WS doit pouvoir confirmer :

- aucune position ouverte
- aucun ordre ouvert
- aucune protection active nécessaire

avant de libérer un lock douteux.

## Hors scope

- Ne pas modifier les règles MTF de chaque profil.
- Ne pas fusionner scalper et scalper_micro.
- Ne pas activer plusieurs profils live tant que cette issue n'est pas terminée.
- Ne pas ajouter de préemption automatique entre profils dans cette première version.

## Acceptance Criteria

- Une clé globale `exchange:market_type:symbol` existe.
- Deux profils différents ne peuvent pas réserver simultanément une ouverture sur le même symbole/exchange/market.
- Le verrou est atomique côté DB.
- Le verrou fonctionne avec plusieurs workers/processes concurrents.
- Le verrou ne dépend pas uniquement du filtre open orders/open positions.
- Une raison `cross_profile_symbol_locked` est retournée/logguée.
- Le lock est libéré sur rejet/annulation/expiration/position fermée.
- Le lock n'est jamais libéré si une position est encore ouverte.
- Un TTL de sécurité existe pour les locks orphelins.
- Une commande de diagnostic permet de lister les locks actifs.
- Des tests prouvent que scalper et scalper_micro ne peuvent pas ouvrir le même symbole en parallèle.

## Commandes de diagnostic attendues

Créer :

```bash
php bin/console app:symbol-lock:list
```

Créer :

```bash
php bin/console app:symbol-lock:release BTCUSDT --exchange=bitmart --market-type=perpetual --reason=manual_investigation
```

La commande de release doit refuser si une position ouverte est détectée, sauf option explicite :

```bash
--force
```

L'option `--force` doit être fortement logguée.

### Tests unitaires

```text
global_symbol_key generation
same profile same symbol blocked by existing active lock
different profile same symbol blocked
different exchange same symbol allowed
different market_type same symbol allowed
different symbol allowed
expired lock can be reclaimed only if no open order/position
```

### Tests repository / DB

```text
worker A réserve bitmart:perpetual:BTCUSDT / scalper
worker B tente bitmart:perpetual:BTCUSDT / scalper_micro
=> worker B reçoit cross_profile_symbol_locked
```

### Tests intégration TradeEntry

```text
MTF scalper READY BTCUSDT
MTF scalper_micro READY BTCUSDT
scalper réserve le lock
scalper_micro est skipped
un seul ordre est soumis
```

### Tests release

```text
order rejected => lock released
order cancelled => lock released
position closed => lock released
unknown status => lock kept
open position exists => lock kept
```

## Documentation attendue

Ajouter :

```text
trading-app/docs/CROSS_PROFILE_SYMBOL_LOCK.md
```

Elle doit expliquer :

- problème
- clé globale
- états bloquants
- états libérables
- commandes diagnostic
- procédure d'incident
- lien avec Temporal schedules

## Notes de sécurité

Cette issue est prioritaire avant d'activer plusieurs profils live sur le même exchange.

Tant que ce lock n'existe pas, éviter :

```text
bitmart scalper live + bitmart scalper_micro live
okx scalper live + okx scalper_micro live
```

Le risque est une double exposition sur le même symbole.

## Prompt Codex recommandé

```text
Tu travailles sur tradingV3.
Objectif : implémenter API-12 — Cross-profile symbol arbitration / global position lock.
Commence par inspecter :
- trading-app/src/TradeEntry/Idempotency/DecisionKeyFactory.php
- trading-app/src/Service/OrderIntentManager.php
- trading-app/src/Entity/OrderIntent.php
- trading-app/src/TradeEntry/Execution/ExchangeExecutionService.php
- trading-app/src/MtfValidator/Service/TradingDecisionHandler.php
- trading-app/src/MtfRunner/Service/MtfRunnerService.php
- repositories liés à OrderIntent, Position, FuturesOrder
- migrations existantes
Problème :
L'idempotence actuelle inclut strategy_profile. Elle protège un même profil, mais ne bloque pas deux profils différents sur le même exchange+market_type+symbol.
Implémente :
1. Une clé globale exchange:market_type:symbol.
2. Un lock atomique DB pour empêcher plusieurs profils d'ouvrir le même symbole.
3. Une raison de skip cross_profile_symbol_locked.
4. Une libération de lock sur rejet/annulation/position fermée.
5. Un TTL de sécurité pour locks orphelins.
6. Des commandes app:symbol-lock:list et app:symbol-lock:release.
7. Tests unitaires + repository + intégration.
8. Documentation CROSS_PROFILE_SYMBOL_LOCK.md.
Contraintes :
- Ne jamais libérer un lock si une position ouverte existe.
- Ne pas se baser uniquement sur le filtre open positions/open orders du runner.
- Le mécanisme doit être robuste avec plusieurs workers parallèles.
- Ne pas activer de préemption entre profils dans cette première version.
- Par défaut : first active intent wins.
```

---

Lié à #126 (API-11 — runtime matrix exchange/profile Temporal schedules).

### PR DESCRIPTION
**Pull request by pivox**
## Summary
- Add a persistent `symbol_execution_lock` table with a PostgreSQL partial unique active lock on `exchange + market_type + symbol`.
- Integrate the lock into `OrderIntentManager::reserveIntent()` for decision-key and no-decision-key reservation paths, returning `cross_profile_symbol_locked` with lock metadata.
- Add guarded release on failed/cancelled intents, position-close events, TTL reclaim without open exposure, and diagnostic list/release commands.
- Document the operational runbook in `trading-app/docs/CROSS_PROFILE_SYMBOL_LOCK.md`.

Closes #127

## Test Plan
- ✅ `php -d error_reporting='E_ALL & ~E_DEPRECATED' ./vendor/bin/phpunit tests/Repository/ExchangeScopedStorageTest.php tests/Command/SymbolLockCommandTest.php` (`22` tests, `95` assertions, `1` skipped)
- ✅ `php -d error_reporting='E_ALL & ~E_DEPRECATED' bin/console lint:container`
- ✅ `php -l` on modified and new PHP files
- ✅ `git diff --cached --check`
- ⚠️ `php -d error_reporting='E_ALL & ~E_DEPRECATED' ./vendor/bin/phpunit` still fails on pre-existing broad-suite issues unrelated to this PR: missing `App\Domain\Ports\Out\IndicatorProviderPort`, missing `App\Domain\Strategy\Service\StrategyBacktester`, missing `App\Domain\Ports\Out\TradingProviderPort`, undefined legacy `Timeframe::H1/H4/M1`, outdated constructor signatures in older tests, and strict coverage-risky tests. Latest run: `539` tests, `1902` assertions, `116` errors, `6` failures, `24` skipped, `144` risky.

### CODE REVIEW

**Comment by pivox** [comment]
@codex please review this PR. Focus on correctness, regressions, architecture boundaries, missing tests, and edge cases. Please post review threads on specific lines when useful, or a PR comment if the feedback is general.

File: trading-app/src/Service/SymbolExecutionLockManager.php [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            } else {
+                return SymbolExecutionLockReservation::blocked($active, $this->lockMetadata($active, $intent));
+            }
+        }
+
+        $lock = new SymbolExecutionLock(
+            exchange: $intent->getExchange(),
+            marketType: $intent->getMarketType(),
+            symbol: $intent->getSymbol(),
+            ownerOrderIntent: $intent,
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Service/SymbolExecutionLockManager.php:73]: Corrigé dans d24053a1. `reserveForIntent()` vérifie maintenant l’exposition locale même sans verrou actif. S’il existe une position ou un ordre ouvert, il crée un verrou actif sans owner et renvoie un blocage avec `blocking_reason=existing_open_exposure`.

**Comment by pivox** [comment]
@codex please review this PR. Focus on correctness, regressions, architecture boundaries, missing tests, and edge cases. Please post review threads on specific lines when useful, or a PR comment if the feedback is general.

File: trading-app/src/Service/SymbolExecutionLockManager.php [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                exchange: $intent->getExchange(),
+                marketType: $intent->getMarketType(),
+                symbol: $intent->getSymbol(),
+                expiresAt: (new \DateTimeImmutable('now', new \DateTimeZone('UTC')))->modify(
+                    sprintf('+%d seconds', $this->defaultTtlSeconds)
+                ),
+            );
+            $lock->setPayload(array_replace($payload, ['source' => 'existing_open_exposure']));
+            $this->entityManager->persist($lock);
+            $this->entityManager->flush();
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Service/SymbolExecutionLockManager.php]: Corrigé dans 016ee5e1. Le chemin `existing_open_exposure` ne flush plus depuis `SymbolExecutionLockManager`; il retourne un verrou synthétique managé avec un flag. `OrderIntentManager` détache d’abord l’`OrderIntent` bloqué et ses protections, puis flush uniquement pour persister ce verrou synthétique. Le test vérifie maintenant que le blocage ne crée aucune ligne `order_intent`.

**Comment by pivox** [comment]
@codex valide cette pr

File: trading-app/src/Entity/SymbolExecutionLock.php [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity(repositoryClass: SymbolExecutionLockRepository::class)]
+#[ORM\Table(name: 'symbol_execution_lock')]
+#[ORM\Index(name: 'idx_symbol_execution_lock_active', columns: ['exchange', 'market_type', 'symbol', 'released_at'])]
+#[ORM\Index(name: 'idx_symbol_execution_lock_owner_intent', columns: ['owner_order_intent_id'])]
+final class SymbolExecutionLock
+{
+    public const STATUS_ACTIVE = 'ACTIVE';
+    public const STATUS_RELEASED = 'RELEASED';
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Entity/SymbolExecutionLock.php:21]: Corrigé dans d67b14b5. `SymbolExecutionLock` déclare maintenant la contrainte unique partielle `ux_symbol_execution_lock_active_symbol` dans la metadata ORM, alignée avec la migration. Le test vérifie les colonnes et `where=released_at IS NULL`; le setup SQLite recrée l’index partiel car DBAL SQLite ignore le `where` des unique constraints générées.

**Comment by pivox** [comment]
@codex please review this PR again after d67b14b5. Focus on correctness, regressions, architecture boundaries, missing tests, and edge cases around symbol locks, reduce-only exits, BitMart order state normalization, and active lock uniqueness.

File: trading-app/src/Service/SymbolExecutionLockManager.php [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        if ($this->hasOpenExposure($active)) {
+            $this->logger->info('symbol_execution_lock.release_skipped_open_exposure', [
+                'exchange' => $active->getExchange(),
+                'market_type' => $active->getMarketType(),
+                'symbol' => $active->getSymbol(),
+                'lock_id' => $active->getId(),
+                'reason' => $reason,
+            ]);
+
+            return false;
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Service/SymbolExecutionLockManager.php]: Addressed by the `OPEN_NUMERIC_STATES` fix in `markOpenOrdersCancelledForIntent` (bbb00cf0). Rows stored with BitMart numeric states `'1'`/`'2'` are now marked as cancelled when `releaseForIntent` runs, so they no longer appear as open exposure on the next `reserveForIntent` call.

File: trading-app/src/Service/OrderIntentManager.php [hunk 5]
```diff
@@ -106,6 +166,40 @@ public function reserveIntent(
 
                 $intent = $this->buildIntent($orderParams, $quantization, $rawInputs);
                 $this->entityManager->persist($intent);
+
+                if ($this->symbolExecutionLockManager instanceof SymbolExecutionLockManager) {
+                    $lockReservation = $this->symbolExecutionLockManager->reserveForIntent(
+                        $intent,
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Service/OrderIntentManager.php:178]: Already addressed: `validateIntent()` calls `symbolExecutionLockManager->releaseForIntent($intent, 'order_intent_validation_failed')` on the failure path.

File: trading-app/src/Service/OrderIntentManager.php [hunk 6]
```diff
@@ -106,6 +166,40 @@ public function reserveIntent(
...
+                            'source' => 'order_intent_reservation',
+                            'decision_key' => $decisionKey,
+                            'strategy_profile' => $intent->getStrategyProfile(),
+                            'strategy_version' => $intent->getStrategyVersion(),
+                        ],
+                    );
+
+                    if ($lockReservation->blocked) {
+                        $this->entityManager->detach($intent);
+                        $connection->commit();
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Service/OrderIntentManager.php:189]: Already addressed: `detachIntentGraph()` iterates `$intent->getProtections()` and detaches each `OrderProtection` before detaching the parent intent.

File: trading-app/src/Service/SymbolExecutionLockManager.php [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+        $this->lockSymbolKey($intent->getExchange(), $intent->getMarketType(), $intent->getSymbol());
+
+        $active = $this->locks->findActive($intent->getExchange(), $intent->getMarketType(), $intent->getSymbol());
+        if ($active instanceof SymbolExecutionLock) {
+            if ($this->canReclaimExpiredLock($active)) {
+                $active->release('expired_reclaimed');
+                $this->entityManager->persist($active);
+                $this->entityManager->flush();
+            } else {
+                return SymbolExecutionLockReservation::blocked($active, $this->lockMetadata($active, $intent));
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Service/SymbolExecutionLockManager.php:46]: Already addressed: `requiresSymbolExecutionLock()` returns `false` for sides 2 and 3, and for `reduce_only`/`reduceOnly` params — close/exit orders bypass the lock entirely.

File: trading-app/src/Repository/FuturesOrderRepository.php [hunk 8]
```diff
@@ -63,4 +66,61 @@ public function findRecentBySymbol(
...
     }
+
+    public function hasOpenOrderForSymbol(string $symbol, ?ExchangeContext $context = null): bool
+    {
+        $count = $this->createQueryBuilder('o')
+            ->select('COUNT(o.id)')
+            ->andWhere('o.exchange = :exchange')
+            ->andWhere('o.marketType = :marketType')
+            ->andWhere('o.symbol = :symbol')
+            ->andWhere('o.status IN (:statuses)')
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Repository/FuturesOrderRepository.php]: Fixed in bbb00cf0. Scalar DQL does not auto-deserialize JSON columns, so `rawData` was returned as a string and the `is_array()` guard silently discarded it. `hasOpenOrderForSymbol` now calls `json_decode` when the value is a string before passing it to `isOpenOrderState`.

File: trading-app/src/Repository/FuturesOrderRepository.php [hunk 9]
```diff
@@ -63,4 +69,132 @@ public function findRecentBySymbol(
...
+        }
+
+        $qb = $this->createQueryBuilder('o')
+            ->update()
+            ->set('o.status', ':cancelled')
+            ->set('o.updatedAt', ':now')
+            ->where('o.exchange = :exchange')
+            ->andWhere('o.marketType = :marketType')
+            ->andWhere('o.symbol = :symbol')
+            ->andWhere('(o.status IN (:statuses) OR o.status IS NULL OR o.status = :blankStatus)')
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Repository/FuturesOrderRepository.php]: Fixed in bbb00cf0. `OPEN_NUMERIC_STATES` (`'1'`, `'2'`) added to the WHERE predicate in `markOpenOrdersCancelledForIntent` so BitMart numeric-state rows are updated alongside the canonical string statuses.

File: trading-app/src/Service/SymbolExecutionLockManager.php [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+        $owner = $active->getOwnerOrderIntent();
+        if ($owner instanceof OrderIntent && $owner->getId() !== $intent->getId()) {
+            return false;
+        }
+
+        if ($allowOpenOrder && $reason === 'order_intent_cancelled') {
+            $this->orders->markOpenOrdersCancelledForIntent($intent);
+        }
+
+        if ($this->hasOpenPosition($active) || (!$allowOpenOrder && $this->hasOpenOrder($active))) {
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Service/SymbolExecutionLockManager.php]: Fixed in bbb00cf0. `releaseForIntent` now checks `hasOpenOrder` for ownerless synthetic locks even when `allowOpenOrder=true`. The bypass applies only when the lock has an explicit owner — meaning we are releasing the lock on behalf of the intent that owns it.

File: trading-app/src/Trading/Listener/SymbolExecutionLockReleaseListener.php [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $marketType = $event->extra['market_type'] ?? $event->extra['marketType'] ?? null;
+        $context = ExchangeContext::fromValues($event->exchange, $marketType);
+
+        $this->symbolExecutionLockManager->releaseForSymbol(
+            $event->positionHistory->symbol,
+            $context,
+            'position_closed',
+            true,
+        );
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Trading/Listener/SymbolExecutionLockReleaseListener.php:31]: Fixed in bbb00cf0. `releaseForSymbol` now accepts an optional `$lockedBefore` timestamp and skips release if the active lock was created after it. `SymbolExecutionLockReleaseListener` passes `positionHistory->closedAt`, so a replayed or delayed `PositionClosedEvent` cannot drop a lock that was reserved after the position actually closed.
